src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -125,7 +125,7 @@ private ConcatImmutableArray<CodeRefactoringProvider> GetProviders(Document docu
             using (Logger.LogBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringsAsync, cancellationToken))
             {
                 var extensionManager = document.Project.Solution.Workspace.Services.GetRequiredService<IExtensionManager>();
-                var tasks = new List<Task<CodeRefactoring?>>();
+                using var _ = ArrayBuilder<Task<CodeRefactoring?>>.GetInstance(out var tasks);
 
                 foreach (var provider in GetProviders(document))
```

Comment:
📝 `GetProviders` returns a collection with a known length. A fixed-size `Task<CodeRefactoring?>` array would be more efficient here.

Reply:
that would result in an array that then needs to be gc'ed.  i'd prefer to we in general pool our temp storage so we can just keep reusing these same smallish temp buffers for all successive calls.